File: js/day70.js
const crypto = require('crypto');
const fs = require('fs');
const readline = require('readline');

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

rl.question('Enter your password: ', (password) => {
    // 1. Generate the SHA-256 hash of the password
    const hash = crypto.createHash('sha256').update(password).digest('hex');
    
    // 2. Save the hash to a text file
    fs.writeFile('password_hash.txt', hash, (err) => {
        if (err) throw err;
        console.log('Success! Password hash saved to password_hash.txt');
        console.log('Hash:', hash);
        rl.close();
    });
});

File: python/day70.py
import hashlib
import getpass

def save_password_hash():
    # 1. Ask for password without echoing it to the terminal screen
    password = getpass.getpass("Enter your password: ")
    
    # 2. Encode string to bytes and generate SHA-256 hash
    hashed_password = hashlib.sha256(password.encode('utf-8')).hexdigest()
    
    # 3. Save the hexadecimal string digest to a text file
    with open("password_hash.txt", "w", encoding="utf-8") as file:
        file.write(hashed_password)
        
    print("Password hash successfully saved to password_hash.txt!")

if __name__ == "__main__":
    save_password_hash()
